eventhubs/data-plane/azure-eventhubs-eph/src/main/java/com/microsoft/azure/eventprocessorhost/PartitionManager.java

```diff
@@ -271,7 +271,7 @@ void onPartitionCheckCompleteTestHook() {
                     throw new FinalException(LoggingUtils.wrapExceptionWithMessage(LoggingUtils.unwrapException(e, null), finalFailureMessage, action));
                 }
             }
-            return (e == null) ? r : null;
+            return r;
```

Comment:
I wish the Java docs https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html#handleAsync-java.util.function.BiFunction- were more explicit. If the previous stage throws (e is not null), then it is not obvious what r might be other than null, but it doesn't actually say that r is null in that case, so I wrote the ternary just to be sure.

Reply:
from what I understand, the e== null here is redundant because e is for sure to be null since previous if statement already throw the exception when e != null.

Reply:
You are correct, sorry! This change is OK.